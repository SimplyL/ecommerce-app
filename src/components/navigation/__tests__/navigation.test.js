import React from 'react';
import { shallow } from 'enzyme';
import Navigation from '../index';
import PATHS from '../../../constants/paths';

describe('Navigation component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Navigation paths={PATHS} cartSize={2} />);
  });

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render navigation with correct number of items', () => {
    expect(wrapper.find('MenuItem')).toHaveLength(2);
  });

  it('should update cart size', () => {
    wrapper.setProps({ cartSize: 4 });
    expect(wrapper.find('CartSize').text()).toEqual('4');
  });
});
