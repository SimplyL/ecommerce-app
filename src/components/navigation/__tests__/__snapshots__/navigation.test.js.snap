// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component should render without crashing 1`] = `
<NavigationContainer>
  <Section>
    <Menu>
      <MenuItem>
        <Link
          to="/"
        >
          Home
        </Link>
      </MenuItem>
      <MenuItem>
        <Link
          to="/cart"
        >
          <CartIcon
            size={34}
          />
          <CartSizeContainer>
            <CartSize>
              2
            </CartSize>
          </CartSizeContainer>
        </Link>
      </MenuItem>
    </Menu>
  </Section>
</NavigationContainer>
`;
